dask/array/tests/test_array_core.py

```diff
@@ -4090,3 +4092,27 @@ def test_from_array_meta():
     meta = sparse.COO.from_numpy(x)
     y = da.from_array(x, meta=meta)
     assert isinstance(y._meta, sparse.COO)
+
+
+def test_compute_chunk_sizes():
+    x = da.from_array(np.linspace(-1, 1, num=50), chunks=10)
+    y = x[x < 0]
+    assert np.isnan(y.shape[0])
+    assert y.chunks == ((np.nan,) * 5,)
+
+    z = y.compute_chunk_sizes()
+    assert y is z
+    assert z.chunks == ((10, 10, 5, 0, 0),)
+    assert len(z) == 25
+
+
+def test_compute_chunk_sizes_dataframe():
+    x = np.linspace(-1, 1, num=5 * 9).reshape(9, 5)
+    ddf = dask.dataframe.from_pandas(pd.DataFrame(x), npartitions=3)
+    y = ddf.values
+    assert np.isnan(y.shape[0]) and y.shape[1] == 5
+    assert y.chunks == ((np.nan, np.nan, np.nan), (5,))
+
+    z = y.compute_chunk_sizes()
+    assert y is z
+    assert z.chunks == ((3, 3, 3), (5,))
```

Comment:
If you want a test like this then you should do something like the following directly within this function.

```python
pd = pytest.importorskip("pandas")
```

However, I'm also not certain what this is trying to test.  It seems to use dask dataframe to create a dask array, but then doesn't use dask dataframe functionality afterwards.  Maybe we just drop this test and the imports above?

Reply:
I implemented this test to make sure the chunk size conversion works properly with 2D arrays. I used a DataFrame because that's the easiest (and most common) way to create a 2D array with unknown chunks.

I create the 2D array manually now without Pandas.